### PR TITLE
React 15.5 migration

### PR DESCRIPTION
**Pull request by adriantoine**
Migration to React 15.5, see https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html.

I've just changed the code to load PropTypes from the `prop-types` package instead of the `react` one, that removes the deprecation warning.

### CODE REVIEW

**Comment by shmidtkirill** [comment]
@adriantoine The latest npm version of `react` and `react-dom` is 15.5.3.

**Comment by adriantoine** [comment]
Indeed, I don't know why I've put `15.5.6` 🤔

Thanks I'll update!

**Comment by andrevenancio** [comment]
Yes please! This warning is driving me crazy.

**Comment by sabarasaba** [comment]
Yes, this is bugging me as well!

File: package.json [hunk 1]
```diff
@@ -69,8 +70,9 @@
     "karma-sourcemap-loader": "^0.3.7",
     "karma-webpack": "^1.8.0",
     "mocha": "^3.1.2",
-    "react": "^15.3.2",
-    "react-dom": "^15.3.2",
+    "prop-types": "~15.5.6",
```
Comment (taion) [inline package.json]: use a caret range here?

File: src/ScrollBehaviorContext.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
 import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (taion) [inline src/ScrollBehaviorContext.js]: alphabetize these by package

**Comment by adriantoine** [comment]
@taion thanks for the view, I've updated the PR.

File: package.json [hunk 3]
```diff
@@ -36,6 +36,7 @@
   "peerDependencies": {
     "history": "^2.0.0 || ^3.0.0",
     "react": "^0.14.0 || ^15.0.0",
+    "prop-types": "^15.0.0",
```
Comment (taion) [inline package.json]: this doesn't need to be a peer dep i don't think?

Comment (adriantoine) [inline package.json]: I wasn't sure where to put it, If it's not a peer, should it be in the normal dependencies?

Comment (taion) [inline package.json]: correct, just a normal dep

Comment (adriantoine) [inline package.json]: Ok, I'll update it

**Comment by adriantoine** [comment]
@taion I've updated the PR.

**Comment by taion** [comment]
Thanks!

**Comment by adriantoine** [comment]
No problem, thanks for merging!

**Comment by andrevenancio** [comment]
Thats for the fast update!

**Pull request merged**